source/DurationPicker.js

```diff
@@ -0,0 +1,733 @@
+			* @public
+			*/
+			hourText: moon.$L('hour'),		// i18n 'HOUR' label in moon.DurationPicker widget
+
+			/**
+			* Optional label for minute
+			*
+			* @type {String}
+			* @default moon.$L('minute')
+			* @public
+			*/
+			minuteText: moon.$L('minute'),		// i18n 'MINUTE' label in moon.DurationPicker widget
+
+			/**
+			* Optional label for second
+			*
+			* @type {String}
+			* @default moon.$L('second')
+			* @public
+			*/
+			secondText: moon.$L('second'),		// i18n 'SECOND' label in moon.DurationPicker widget
+
+			/**
+			* The template format of the picker
+			*
+			* @type {String}
+			* @default ''
+			* @public
+			*/
+			template: '',
+
+			/**
+			* Enable and disable the timer feature of duration picker
+			*
+			* @type {Boolean}
+			* @default true
+			* @public
+			*/
+			enableTimer: true
+		},
+
+		/**
+		* @type {Boolean}
+		* @default false
+		* @private
+		*/
+		countdown: false,
+
+		/**
+		* @type {String}
+		* @default null
+		* @private
+		*/
+		timer: null,
+
+		/**
+		* @type {Object}
+		* @default {}
+		* @private
+		*/
+		valueArray: {},
+
+		/**
+		* @type {Array}
+		* @default []
+		* @private
+		*/
+		templateArray: [],
+
+		/**
+		* Optional label for Start Button
+		*
+		* @type {String}
+		* @default moon.$L('Start')
+		* @private
+		*/
+		startConent: moon.$L('Start'),
+
+		/**
+		* Optional label for Pause Button
+		*
+		* @type {String}
+		* @default moon.$L('Pause')
+		* @private
+		*/
+		pauseConent: moon.$L('Pause'),
+
+		/**
+		* @type {Boolean}
+		* @default true
+		* @private
+		*/
+		pauseDisabled: true,
+
+		/**
+		* @private
+		*/
+		observers: {
+			'changeCountdown': ['countdown']
+		},
+
+		/**
+		* @private
+		*/
+		components: [
+			{name: 'headerWrapper', kind: 'moon.Item', classes: 'moon-date-picker-header-wrapper', onSpotlightFocus: 'headerFocus', ontap: 'expandContract', components: [
+				// headerContainer required to avoid bad scrollWidth returned in RTL for certain text widths (webkit bug)
+				{name: 'headerContainer', classes: 'moon-expandable-list-item-header moon-expandable-picker-header moon-expandable-datetime-header', components: [
+					{name: 'header', kind: 'moon.MarqueeText'}
+				]},
+				{name: 'currentValue', kind: 'moon.MarqueeText', classes: 'moon-expandable-picker-current-value'}
+			]},
+			{name: 'drawer', kind: 'enyo.Drawer', resizeContainer: false, classes: 'moon-expandable-list-item-client moon-duration-picker-item', components: [
+				{name: 'client', kind: 'enyo.Control', classes: 'enyo-tool-decorator', onSpotlightLeft: 'closePicker', onSpotlightSelect: 'closePicker'},
+				{name: 'buttonPlaceHolder', components: [
+					{kind: 'moon.Button', name: 'startButton', content: this.startConent, small: true, ontap: 'startTapped'},
+					{kind: 'moon.Button', name: 'pauseButton', content: this.pauseConent, small: true, ontap: 'pauseTapped'}
+				]}
+			]}
+		],
+
+		/**
+		* @private
+		*/
+		bindings: [
+			{from: '.disabled', to: '.$.headerWrapper.disabled'},
```

Comment:
Hi @anishlg, per @Djspaceg's previous comment, we can and should remove the leading "." before each of these binding paths as they are no longer necessary, and the removal of the "." prefix will speed things up.